tests/views_integration/README.md

```diff
@@ -322,4 +322,92 @@ This second test example shows what is, probably, the main feature of data-drive
 @todo @paulmskim
 
 ### Other Testing
-@todo @bordoni
+
+Besides all the tests mentioned above due to the multiple moving parts of the software is important to make sure we test unwanted behavior to make sure they are not being "fixed" in a future bug report.
+
+The test below is dealing with the multiple possibilities of the template bootstraping, and how it is based on an admin setting, which can be changed in a lot of unpredictable ways, so we bake into our integration testing a way to make sure even if those bad values are pulled from the database we still continue to behave as expoected.
+
+```php
+<?php
+namespace Tribe\Events\Views\V2;
+
+class TemplateBootstrapTest extends \Codeception\TestCase\WPTestCase {
+	private function make_instance() {
+		return new Template_Bootstrap();
+	}
+
+	/**
+	 * @test
+	 */
+	public function it_should_be_instantiatable() {
+		$sut = $this->make_instance();
+
+		$this->assertInstanceOf( Template_Bootstrap::class, $sut );
+	}
+
+	public function base_template_options() {
+		return [
+			'invalid' => [
+				'foo',
+				'event',
+			],
+			'numeric' => [
+				2,
+				'event',
+			],
+			'default' => [
+				'default',
+				'page',
+			],
+			'empty_string' => [
+				'',
+				'event',
+			],
+			'numeric_zero' => [
+				0,
+				'event',
+			],
+			'null' => [
+				null,
+				'event',
+			],
+			'boolean_false' => [
+				false,
+				'event',
+			],
+			'boolean_true' => [
+				false,
+				'event',
+			],
+			'slug_event' => [
+				'event',
+				'event',
+			],
+			'slug_page' => [
+				'page',
+				'event',
+			],
+		];
+	}
+
+	/**
+	 * @test
+	 * @dataProvider base_template_options
+	 */
+	public function should_only_allow_permitted_values_on_base_template_option( $input, $expected ) {
+		tribe_update_option( 'tribeEventsTemplate', $input );
```

Comment:
```suggestion
		// The bootstrapping depends on this option, rather than arguments, here we "mock" the test context.
		tribe_update_option( 'tribeEventsTemplate', $input );
```